src/Drips.sol

```diff
@@ -666,22 +698,23 @@ library Drips {
     /// @param amtPerSec The dripping rate
     function _addDelta(
         mapping(uint32 => AmtDelta) storage amtDeltas,
-        uint32 cycleSecs,
-        uint32 timestamp,
+        uint256 cycleSecs,
+        uint256 timestamp,
         int256 amtPerSec
     ) private {
         unchecked {
-            AmtDelta storage amtDelta = amtDeltas[_cycleOf(timestamp, cycleSecs)];
+            AmtDelta storage amtDelta = amtDeltas[_cycleOf(uint32(timestamp), uint32(cycleSecs))];
             int256 thisCycleDelta = amtDelta.thisCycle;
             int256 nextCycleDelta = amtDelta.nextCycle;
 
             // In order to set a delta on a specific timestamp it must be introduced in two cycles.
             // The cycle delta is split proportionally based on how much this cycle is affected.
             // The next cycle has the rest of the delta applied, so the update is fully completed.
-            uint32 nextCycleSecs = timestamp % cycleSecs;
-            uint32 thisCycleSecs = cycleSecs - nextCycleSecs;
-            thisCycleDelta += int256(uint256(thisCycleSecs)) * amtPerSec;
-            nextCycleDelta += int256(uint256(nextCycleSecs)) * amtPerSec;
+            int256 amtPerCycle = (amtPerSec * int256(cycleSecs)) / int256(AMT_PER_SEC_MULTIPLIER);
```

Comment:
First of all we don't use `cycleSecs = 1` or `5`, it'd be pointless to have cycles that short. Rounding down by up to `10^-6` USDC once per week or per month is much less of a problem.

If the `cycleSecs` is `5` seconds, you can choose any number of full `10^-6` USDCs to be dripped over a 5-second cycle. You're right about the limits of precision, for the extremely low `5` second `cycleSecs` and a low precision token like `USDC` these limits are easy to show.

Reply:
L2s can't get cycles much shorter than L1, because usually despite being cheaper they have more constrained block gas limits. A 5-second cycle is insane on every chain, it'd be better to just have a 1-to-1 dripping protocol and iterate over all the senders when collecting.

> Did we mention that in our design document?

We had an issue https://github.com/radicle-dev/drips-contracts/issues/159#issuecomment-1155456314 which noted about this solution that:
> When dripping over a boundary between cycles, round down while crossing it.

Where are you going with this?

Reply:
> When dripping over a boundary between cycles, round down while crossing it.


Yeah, but i missed the implication that smaller cycles will impact the precision of the fractions.

Reply:
It was just more a note to myself that I missed that until now. No, blocker.

Reply:
I really wouldn't worry too much about extremely short cycles UX, anything below ~1 day is going to be barely usable anyway. The rounding is deterministic too, it can be accounted for when setting up a drip.